packages/jsii-diff/lib/classes-ifaces.ts

```diff
@@ -0,0 +1,203 @@
+import reflect = require('jsii-reflect');
+import log4js = require('log4js');
+import { Analysis, FailedAnalysis, isSuperType } from './type-analysis';
+import { ComparisonContext, shouldInspect } from './types';
+
+const LOG = log4js.getLogger('jsii-diff');
+
+/**
+ * Compare two class types
+ *
+ * We require that all stable properties and methods on the original are
+ * present on the new type, and that they match in turn.
+ */
+export function compareClass(original: reflect.ClassType, updated: reflect.ClassType, context: ComparisonContext) {
+  if (updated.abstract && !original.abstract) {
+    context.mismatches.report(original, 'has gone from non-abstract to abstract');
+  }
+
+  for (const [origMethod, updatedMethod] of memberPairs(original, original.methods, updated, context)) {
+    compareMethod(original, origMethod, updatedMethod, context);
+  }
+
+  for (const [origProp, updatedProp] of memberPairs(original, original.properties, updated, context)) {
+    compareProperty(original, origProp, updatedProp, context);
+  }
+
+  // You cannot have added abstract members to the class, as they are
+  // an added burden on potential subclass implementors.
+  // Only for types that are explicitly marked as intended to be subclasses by customers.
+  if (subclassableType(original)) {
+    checkNoNewMembers(original, updated, context);
+  }
+}
+
+export function compareInterface(original: reflect.InterfaceType, updated: reflect.InterfaceType, context: ComparisonContext) {
```

Comment:
I would prefer to see a separate function for structs (esp given we will eventually model
Them as a different entity). In fact, I think classes and interfaces can probably share the same
Comparer. Also, see my comment about making output structs explicit so we don’t need the heuristic here